src/app/locales/vi_VN.ts

```diff
@@ -0,0 +1,513 @@
+    action: 'Ở trên thuộc tính `routerLink` của Angular Router, `preserveQueryParams` đã được xoá bỏ, dùng `queryParamsHandling="preserve"`.'
+  },
+  'routerlink queryParams typing': {
+    action: 'Nếu như bạn đang truy xuất đến giá trị `queryParams`, `fragment`, hoặc `queryParamsHandling` trên `routerLink`, có thể bạn sẽ cần phải thả lỏng kiểu giá trị một chút để có thể chấp nhận cả `undefined` và `null`.'
+  },
+  'viewencapsulation native removed': {
+    action: 'Tuỳ chỉnh View Encapsulation trên Component `ViewEncapsulation.Native` đã được xoá bỏ. Dùng `ViewEncapsulation.ShadowDom`. `ng update` sẽ thực hiện việc thay đổi này cho bạn.'
+  },
+  'ICU expressions typechecked': {
+    action: 'Nếu bạn sử dụng i18n, những biểu thức bên trong Internaltional Components for Unicode (ICUs) biểu thức sẽ được kiểm tra kiểu dữ liệu. Điều này có thể dẫn đến lỗi khi biên dịch code.'
+  },
+  'forms validators asyncValidators typing': {
+    action: 'Directives ở trong `@angular/forms` từng có kiểu dữ liệu là `any[]` cho `validators` và `asyncValidators`. Các tham số này từ bây giờ đã được cung cấp kiểu dữ liệu phù hợp, nếu như code của bạn phụ thuộc vào kiểu dữ liệu `any[]` cho các tham số này, có thể bạn sẽ cần phải cập nhật lại code để cải thiện kiểu dữ liệu.'
+  },
+  'forms AbstractFormControl': {
+    action: 'Nếu bạn sử dụng Angular Forms, kiểu dữ liệu của `AbstractFormControl.parent` bây giờ bao gồm cả `null`. `ng update` sẽ thực hiện việc thay đổi này cho bạn, nhưng trong trường hợp hiếm, code của bạn kiểm tra xem `AbstractFormControl.parent` có `undefined` hay không bằng `===`, bạn cần phải dùng `=== null` vì `parent` bây giờ sẽ được khởi tạo bằng giá trị `null` thay vì là được để `undefined` như trước.'
+  },
+  'platform-webworker': {
+    action: 'Thư viện hiếm khi được sử dụng, `@angular/platform-webworker` và `@angular/platform-webworker-dynamic` đã được khai tử ở phiên bản 8, và giờ đã được xoá bỏ hoàn toàn. Thực thi một số thành phần của Angular bên trong môi trường của một Web Worker là một thí nghiệm mà không bao giờ hoạt động chính xác trong nhiều trường hợp thông dụng. Angular vẫn có hỗ trợ cho [Web Workers](https://angular.io/guide/web-worker).'
+  },
+  'v11 slice pipe typing': {
+    action: '`slice` Pipe giờ sẽ trả về `null` nếu như giá trị đầu vào là `undefined`. Điều này khiến cho `slice` Pipe sẽ thống nhất với phần lớn các Pipes hiện tại.'
+  },
+  'v11 keyvalue typing': {
+    action: '`keyvalue` Pipe đã được chỉnh sửa cho trường hợp nếu như đối tượng (object) đầu vào có keys là kiểu dữ liệu số (number), kiểu dữ liệu của giá trị trả về từ Pipe sẽ là giá trị chuỗi của những keys này (string representation). Điều này đó giờ là như vậy rồi nhưng code của `keyvalue` Pipe được cập nhật để phản ánh đúng mà thôi. Hãy cập nhật những nơi sử dụng pipe này nếu như những nơi này đang phụ thuộc vào kiểu dữ liệu không đúng trước đó. Chú ý là điều này không ảnh hưởng đến những trường hơp mà giá trị đầu vào là `Map`.'
```

Comment:
For this sentence "Điều này đó giờ là như vậy rồi nhưng code của `keyvalue` Pipe được cập nhật để phản ánh đúng mà thôi", I would suggest "Từ trước đến giờ nó đã hoạt động như vậy, nhung code của `keyvalue` Pipe được cập nhật để phản ánh đúng mà thôi."